### PR TITLE
#144 upload videos

### PR DESCRIPTION
**Pull request by pdvorchik**
This is a first pass at closing #144.

The upload page could likely use a bit more styling and content, and I haven't done anything with hashtags on the events yet (leaving that for a different issue).

To test/use (steps 3-5 are also for you guys @nathanmwilliams and jimmy):
1) Go to http://localhost:3000/#upload-video
2) Follow the instructions the page
3) Login to the youtube account for townhallprojectcontact@gmail.com (ask myself or @meganrm for the password).
4) Vet any movies that have been uploaded.  Make sure they're not porn or hate speech or anything else.
5) Download the video from youtube and upload it to our real THP youtube account
OR
5b) Go to the real THP youtube account and "remix" the video to move it over to that channel.

**NB:  We really need a second official youtube account to disseminate videos.  A hacker could upload videos and force them to be visible on this account.    Meaning that this account should only ever be used for vetting, and NEVER for our actual PR or sharing.**

Screenshots:
image
image
image
image

image
image
image
image

image

### CODE REVIEW

**Comment by nathanmwilliams** [comment]
Possible to add some copy in/around the description like "Please highlight timecode with questions, answers, or other powerful moments we should share with others."

**Comment by pdvorchik** [comment]
Sure will do.  I'll try and move the oauth step to stage 2 as well, so they can see the form from the beginning.  Might mean there are times when they can see the form but uploading gives an error but it's google we're talking about so it should work 99.9% of the time.

**Comment by pdvorchik** [comment]
image

Descriptions updated.  Tried a few times to get the workflow working where auth happened on upload but it's still a bit patchy so I'll hold off for now.

File: index.html [hunk 1]
```diff
@@ -381,7 +382,48 @@ <h2 class="text-secondary-dark">About Town Hall Project</h2>
           </div>
 
       </div>
-      <div role="tabpanel" class="tab-pane " id="privacy-policy">
+      <div role="tabpanel" class="tab-pane page" id="upload-video">
+        <section class="container">
+          <h1 class="text-primary text-center">Share your Town Hall Videos</h2>
+          <div class="col-sm-8 col-sm-offset-2 upload-video-stage-1">
+            <h3>Have a video from a town hall you want to share?</h3>
+            <button class="btn btn-primary btn-light-background upload-video-begin">Click to begin uploading</button>
```
Comment (meganrm) [inline index.html]: I would right justify this button

File: index.html [hunk 2]
```diff
@@ -381,7 +382,48 @@ <h2 class="text-secondary-dark">About Town Hall Project</h2>
...
+                <em>By uploading a video you grant the Town Hall Project permission to use the video and share it with our progressive partners.  You also certify that you own all rights to the content or that you are authorized by the owner to make the content publicly available.</em>
+              </div>
+              <button name="button" class="btn btn-primary btn-light-background btn-lg upload-video-upload">Upload my video</button>
+            </form>
+          </div>
+          <div class="col-sm-8 col-sm-offset-2 upload-video-stage-4 is_hidden">
+            <h3>
+              <div>Upload in progress:</div>
+              <div><progress id="upload-video-progress" max="1" value="0"></progress> <span id="upload-video-percent-transferred"></span>%</div>
+              <div><span id="upload-video-bytes-transferred"></span>/<span id="upload-video-total-bytes"></span> bytes</div>
```
Comment (meganrm) [inline index.html]: this ends up being an excessively large number.

**Comment by nathanmwilliams** [comment]
I think we can change "with our progressive partners" to "with the public." Any of these that get approved will be posted publicly--we'll be making this available to progressive partners but also anyone else who wants to view.

**Comment by pdvorchik** [comment]
Thanks for the feedback guys!

@nathanmwilliams I've updated the copy.

Megan:  The right align stuff looked weird on desktop.  I pushed up a version where the button and text are centered, let me know what you think.  Can still do right if you prefer but figured I'd offer an alternative. I got rid of the byte counter and put in approximate minutes and seconds remaining instead.

There's definitely room for improvement on both screens stylistically.  Most of the screens look pretty empty.  I'd love to do some tweaks later (have it auth when you attempt to upload so we can get rid of the first screen all together, etc), but this should get us like 95% of the way there.

image
image


image
image

**Comment by nathanmwilliams** [comment]
Hey @pdvorchik, I just talked to an activist who mentioned off-hand she uploaded a video using our tool a couple months back. But I don't see it on this channel. Then I tried a video upload test today and don't see it either. 

Any idea what's up?

**Comment by pdvorchik** [comment]
@nathanmwilliams Are you using the "THP Video Submissions Partners" account (you can access it from the THP Video Submissions account)?  I know it's a bit confusing, but that was needed to give other partners access to the account as discussed before.

I believe the activist uploaded this one:
Senator Ron Wyden, Oregon 12/3/17 
Dec 14, 2017 3:56 PM
https://www.youtube.com/edit?o=U&video_id=paCD6Dch_yE

And you uploaded this one:
test HD 
Jan 12, 2018 5:34 PM
https://www.youtube.com/edit?o=U&video_id=ViGE8qLERzk

image

**Comment by nathanmwilliams** [comment]
Ah, gotcha. Thanks for walking me through this again. These youtube interfaces are so un-intuitive...

**Comment by pdvorchik** [comment]
Yeah they really are.  I knew what I was looking for and it still took me a bit to find it.

**Pull request merged**